packages/bodiless-documentation/doc/Development/Guides/DesignElementConcepts.md

```diff
@@ -1,329 +1,327 @@
+    };
+    ```
 
-   ```ts
-   export * from './components/RichText';
-   ```
+01. Now export these tokens from your package by creating the following files:
 
-Rebuild your package (execute `npm run build` from the package directory or
-`npm run build:packages` from outside it), and use these tokens in the Gallery.
-In `sites/mysite/src/data/pages/gallery/index.tsx`, add these imports:
+    01. `/packages/mysite/src/components/Element/tokens/index.ts`:
 
-```ts
-import { withNodeKey } from '@bodiless/core';
-import { RichText } from '@bodiless/richtext-ui';
-import { withPlaceholder } from '@bodiless/components';
-import { mysiteElement, mysiteRichText } from 'mysite';
-```
+        ```ts
+        import mysiteElement from './mysiteElement';
 
-and then change:
+        export default mysiteElement;
+        ```
 
-```ts
-const Body = withSimpleEditor('body', 'Body')(Div);
-```
+    01. `/packages/mysite/src/components/Element/index.ts`:
 
-to
+        ```ts
+        import mysiteElement from './tokens';
 
-```ts
-const Body = as(
-  mysiteRichText.Simple,
-  withPlaceholder('Body'),
-  withNodeKey('body'),
-)(RichText);
-```
+        export { mysiteElement };
+        ```
 
-Similarly, in `sites/mysite/src/data/pages/Gallery/CaptionedImage.tsx`, add the imports
-and change
+    01. `/packages/mysite/src/index.ts`:
 
-```ts
-const Body = as(
-  addClasses('text-white bg-black flex-grow p-2'),
-  withSimpleEditor('caption', 'Caption')
-)(Div);
-```
+        ```ts
+        export * from './components/Element';
+        ```
 
-to
+    ?> **Note:** There are many ways to organize exports from component packages. Choosing a
+    standard pattern such as the one described above will help keep your code organized and make it
+    easier to maintain. Following this pattern will also allow you to take advantage of some
+    BodilessJS tools for extending and optimizing your design system, like [Token
+    Shadowing](/VitalDesignSystem/Components/VitalElements/Shadow) and [Static
+    Replacement](/VitalDesignSystem/Components/VitalElements/StaticReplacement).
 
-```ts
-const Body = as(
-  mysiteRichText.Simple,
-  withPlaceholder('Caption'),
-  withNodeKey('caption'),
-  addClasses('text-white bg-black flex-grow p-2'),
-)(RichText);
-```
+### 1b. Update Your Site to Use the New Package and Element Component
+
+01. Add your package as a dependency in `/sites/mysite/package.json`:
+
+    ```json
+    "dependencies:" {
+      ...,
+      "mysite": "^0.0.0",
+    }
+    ```
+
+01. Run `npm run setup` from the root of your repository to build your package and link it to your
+    site.
+
+01. Now let's use these tokens in our Gallery page. Add the following import to
+    `/sites/mysite/src/data/pages/gallery/index.tsx`:
+
+    ```ts
+    import { mysiteElement } from 'mysite';
+    ```
+
+    And replace the definition of the `PrimaryHeader` component:
+
+    ```ts
+    const PrimaryHeader = as(
+      asEditable('title', 'Title'),
+      mysiteElement.H1,
```

Comment:
For this, and other tokens we create in this tutorial, I receive the following error from TypeScript:

```
Argument of type 'TokenSpec<DesignableComponents, { Core: {}; Schema: {}; Layout: {}; Theme: {}; Meta: {}; Editors: {}; }>' is not assignable to parameter of type 'Token'.
  Type 'TokenSpec<DesignableComponents, { Core: {}; Schema: {}; Layout: {}; Theme: {}; Meta: {}; Editors: {}; }>' is not assignable to type 'TokenSpec<any, {}>'.
    Type 'TokenSpec<DesignableComponents, { Core: {}; Schema: {}; Layout: {}; Theme: {}; Meta: {}; Editors: {}; }>' is not assignable to type 'Required<FinalDomains<any, {}>>'.
      Types of property 'Compose' are incompatible.
        Type '{ [key: string]: ComposedToken<DesignableComponents, { Core: {}; Schema: {}; Layout: {}; Theme: {}; Meta: {}; Editors: {}; }>; }' is not assignable to type '{ [key: string]: ComposedToken<any, {}>; }'.
          'string' index signatures are incompatible.
            Type 'ComposedToken<DesignableComponents, { Core: {}; Schema: {}; Layout: {}; Theme: {}; Meta: {}; Editors: {}; }>' is not assignable to type 'ComposedToken<any, {}>'.
              Type 'TokenSpec<DesignableComponents, { Core: {}; Schema: {}; Layout: {}; Theme: {}; Meta: {}; Editors: {}; }>' is not assignable to type 'ComposedToken<any, {}>'.
```